src/ray/raylet/node_manager.cc

```diff
@@ -1133,10 +1133,22 @@ void NodeManager::HandleTaskReconstruction(const TaskID &task_id) {
       /*failure_callback=*/
       [this](ray::gcs::AsyncGcsClient *client, const TaskID &task_id) {
         // The task was not in the GCS task table. It must therefore be in the
-        // lineage cache. Use a copy of the cached task spec to re-execute the
-        // task.
-        const Task task = lineage_cache_.GetTask(task_id);
-        ResubmitTask(task);
+        // lineage cache.
+        if (!lineage_cache_.ContainsTask(task_id)) {
```

Comment:
I think this is not the way to solve it.

Reply:
I left a TODO explaining that we should revert this change once #2608 is fixed. I don't think we should block this PR on #2608 though.

Reply:
@stephanie-wang  That's a nice solution also. 
btw, does it still hang after ur this change?

Reply:
Nope! Although there are many error messages (and will be for any job with tasks that take some time to complete) logged.